sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/InstrumentationPolicy.java

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.http.policy;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.SpanKind;
+import com.azure.core.util.tracing.StartSpanOptions;
+import com.azure.core.util.tracing.Tracer;
+import reactor.core.CoreSubscriber;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Operators;
+import reactor.core.publisher.Signal;
+
+import static com.azure.core.util.tracing.Tracer.DISABLE_TRACING_KEY;
+
+public class InstrumentationPolicy implements HttpPipelinePolicy {
+    static final String HTTP_USER_AGENT = "http.user_agent";
+    static final String HTTP_METHOD = "http.method";
+    static final String HTTP_URL = "http.url";
+    static final String HTTP_STATUS_CODE = "http.status_code";
+    static final String SERVICE_REQUEST_ID_ATTRIBUTE = "serviceRequestId";
+    static final String CLIENT_REQUEST_ID_ATTRIBUTE = "requestId";
+    private static final String REACTOR_HTTP_TRACE_CONTEXT_KEY = "instrumentation-context-key";
+    private static final String SERVICE_REQUEST_ID_HEADER = "x-ms-request-id";
+    private static final String CLIENT_REQUEST_ID_HEADER = "x-ms-client-request-id";
```

Comment:
Let's make these `HttpHeaderName`s so this policy takes benefit of their performance improvement when interacting with `HttpHeaders`.